openedx/core/djangoapps/discussions/serializers.py

```diff
@@ -38,14 +41,26 @@ def to_internal_value(self, data: dict) -> dict:
         }
         return payload
 
+    def to_representation(self, instance):
+        representation = super().to_representation(instance)
+        if not self.context.get('pii_sharing_allowed'):
+            representation.pop('pii_share_username')
+            representation.pop('pii_share_email')
+        return representation
+
     def update(self, instance: LtiConfiguration, validated_data: dict) -> LtiConfiguration:
         """
         Create/update a model-backed instance
         """
         instance = instance or LtiConfiguration()
         instance.config_store = LtiConfiguration.CONFIG_ON_DB
+        pii_sharing_allowed = self.context.get('pii_sharing_allowed', False)
         if validated_data:
             for key, value in validated_data.items():
+                if key.startswith('pii_') and not pii_sharing_allowed:
```

Comment:
are there any other fields other than `pii_share_username` and `pii_share_email` starting with `pii_` for this check?

Reply:
Nope, but the idea is that if there are any in the future, this will cover it.